### PR TITLE
feat: extend the referral view on the user profile

### LINKED ISSUE #1182: Extend referral view design changes
Users should be able to see whom of their referrals submitted to the challenge, see the submission and see if the submission was successful or not. We need that, especially for the ICP hubs, to reduce Nathania's workload. 

- [ ] Show the name of the challenge they submitted with a link to the challenge.
- [ ] If the submission has not been evaluated yet, write evaluation pending.
- [ ] If the submission has been evaluated, and if the submission was successful (passed the threshold and earned a certificate), that the user passed the challenge.
- [ ] -If the submission has been evaluated, and if the submission was NOT successful (didn't pass the threshold and earned a certificate), the user didn't pass the challenge.

Take wordings and design from Figma:
https://www.figma.com/file/0BRR6Ch2qbOEUT4zIdmmzg/Dacade-2?type=design&node-id=586-8149&mode=design&t=A8zd7z2peKfMdFMN-0

Here is the current view of the referrals page.
https://dacade.org/profile/referrals

### PR DESCRIPTION
**Pull request by serapieTuyishime**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

This is aiming to implement the design for showing referred users to the user, and their current submissions status.

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

closes #1182 

## Other Relevant Information

Provide any other important details below.

## Summary by CodeRabbit

- **New Features**
	- Added new status descriptions and translations for referral challenge evaluations across multiple languages (Bulgarian, English, Spanish, Croatian).
	- Introduced new components for profile referrals management, including `ProfileOverviewSection` and `ReferralsList`.
	- Enhanced user experience in referral details display and management on profile pages.

- **Bug Fixes**
	- Updated the layout and logic for displaying referrals, including improved empty state handling.

- **Documentation**
	- Updated language files to refine and add new translations related to referrals.

- **Refactor**
	- Refactored status evaluation and date formatting functions in the referral component.
	- Improved data access logic in the referrals service function.

- **Style**
	- Adjusted layout and styling for the referral component and badges based on the type of reward.

### CODE REVIEW

File: src/components/cards/profile/Referral.tsx [hunk 1]
```diff
@@ -25,54 +27,58 @@ export default function Referral({ referral }: ReferralProps): ReactElement {
...
-                </li>
-              )}
-
-              {referral.submission && referral.submission.metadata && referral.submission.metadata.evaluation && (
-                <li>
-                  <span className="ml-5">{t("referrals.submission.evaluation")}</span>
-                  <span className="font-bold">
-                    {referral.submission.metadata.evaluation.points}/{referral.submission.metadata.evaluation.totalPoints}
+            <ul className="pb-1 font-light leading-loose text-gray-700 grid divide-y-2 space-y-4 divide-gray-200">
+              {referral?.submissions.length ? referral.submissions.map((_submission) => (
```
Comment (Jonath-z) [inline src/components/cards/profile/Referral.tsx:53]:
```suggestion
              {referral?.submissions.length ? referral.submissions.map((submission) => (
```

File: src/components/cards/profile/Referral.tsx [hunk 2]
```diff
@@ -25,54 +27,58 @@ export default function Referral({ referral }: ReferralProps): ReactElement {
...
+                <li key={_submission.id} className="grid md:flex justify-between pt-4 gap-2 md:gap-0">
+                  <span className="grid gap-2">
+                    <span>
+                      {status(_submission.metadata.evaluation)} <Link href={`${_submission.challenge.name}`} className="font-bold underline text-base- underline-offset-2">{_submission.challenge.name} {t("referrals.submission.challenge")}</Link>
```
Comment (Jonath-z) [inline src/components/cards/profile/Referral.tsx:57]:
```suggestion
                      {status(_submission.metadata.evaluation)} <Link href={_submission.challenge.name} className="font-bold underline text-base- underline-offset-2">{_submission.challenge.name} {t("referrals.submission.challenge")}</Link>
```

**Review by Samantha-KY — APPROVED** [review]

File: src/components/ui/SideNavigation/index.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ export default function SideNavigation({ items, colors, children }: { items: Ite
   const { t } = useTranslation();
   return (
     <ThemeWrapper colors={colors}>
-      <ul className="relative">
+      <ul className="relative bg-red-400">
```
Comment (Odile078) [inline src/components/ui/SideNavigation/index.tsx:19]: should this `background` be added?

**Comment by Odile078** [comment]
A few changes on the view,

- The gap between a coin and the amount seems bigger than 
kingokonn

- There layout mismatch on an iPad,
Screenshot 2024-04-15 at 9 20 15 AM

File: src/components/cards/profile/Referral.tsx [hunk 4]
```diff
@@ -25,54 +28,58 @@ export default function Referral({ referral }: ReferralProps): ReactElement {
...
-                <li>
-                  <span className="ml-5">{t("referrals.submission.evaluation")}</span>
-                  <span className="font-bold">
-                    {referral.submission.metadata.evaluation.points}/{referral.submission.metadata.evaluation.totalPoints}
+            <ul className="pb-1 font-light leading-loose text-gray-700 grid divide-y-2 space-y-4 divide-gray-200">
+              {referral?.submissions?.length ? referral.submissions.map((submission) => (
+                <li key={submission.id} className="grid md:flex justify-between pt-4 gap-2 md:gap-0">
+                  <span className="grid gap-2">
+                    <span>
+                      {status(submission.metadata.evaluation)} <Link href={`/communities/${submission.community.slug}/challenges/${submission.challenge.id}/submissions/${submission.id}`} className="font-bold underline text-base- underline-offset-2">{submission.challenge.name} {t("referrals.submission.challenge")}</Link>
```
Comment (Jonath-z) [inline src/components/cards/profile/Referral.tsx:58]: `{`/communities/${submission.community.slug}/challenges/${submission.challenge.id}/submissions/${submission.id}` can you build on top, having it inline does not make readable

**Review by Jonath-z — APPROVED** [review]

**Review by Odile078 — APPROVED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**